doc/ipython-notebooks/link_checker.py

```diff
@@ -0,0 +1,63 @@
+import json
+import re
+import subprocess
+import sys
+
+
+#requirement is python >= 3.5
+
+def Find(cell_contents):
+    text = cell_contents[0]
+    urls = re.findall('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', text)
+    return urls   
+
+def link_finder(file_name):
+    
+    links = []
+    f = open(file_name)
+    s = f.read()
+    input_file = json.loads(s)
+    #convert given file to python dictionary fromat
+    cells_count = len(input_file['cells'])
+    for i in range(cells_count):
+        #Assuming all links are in markdown cells
+        if(input_file['cells'][i]['cell_type'] == 'markdown'):
+            urls_in_current_cell = Find(input_file['cells'][i]['source'])
+            links.extend(urls_in_current_cell)
+    return links       
+
+def link_checker(link_address):
+    process = subprocess.Popen(['curl','-s', '-o','/dev/null','-I','-w','%{http_code}',link_address],stdout=subprocess.PIPE, stderr=subprocess.PIPE,universal_newlines=True)
```

Comment:
rather that CURL can you not use python's `request` package and check the return code?

Reply:
I actuallt taklked about it with @vigsterkr and he guided me to use curl
Could definitely do it with python requests package.
Will get on it

Reply:
ah ok, well let's see what @vigsterkr thinks. It's just a suggestion :)

Reply:
could you please confirm?

Reply:
curl was about the idea when it was a shellscript... in case of python i dont see the reason to call out with a subprocess and call curl. as @gf712 use request package or any basic python functionality that fetches an url... even urllib (which is by default part of python) could work.

Reply:
OH that makes sense. My bad. Will get on it right now.Altering the link_checker function :)